verilog/tools/kythe/kythe_facts_extractor.cc

```diff
@@ -0,0 +1,154 @@
+      break;
+    case IndexingFactType::kModulePortOutput:
+      break;
+  }
+
+  const AutoPop p(&facts_tree_context_, &node);
+  for (const verible::VectorTree<IndexingNodeData>& child : node.Children()) {
+    Visit(child);
+  }
+}
+
+void KytheFactsExtractor::ExtractFileFact(const IndexingFactNode& node) {
+  const VName file_vname(file_path_, "", "", "");
+
+  std::cout << Fact(file_vname, kFactNodeKind, kNodeFile);
+  std::cout << Fact(file_vname, kFactText, node.Value().Anchors()[1].Value());
+}
+
+void KytheFactsExtractor::ExtractModuleFact(const IndexingFactNode& node) {
+  const auto& anchors = node.Value().Anchors();
+  const VName module_vname(file_path_,
+                           CreateModuleSignature(anchors[0].Value()));
+
+  std::cout << Fact(module_vname, kFactNodeKind, kNodeRecord);
+  std::cout << Fact(module_vname, kFactSubkind, kSubkindModule);
+  std::cout << Fact(module_vname, kFactComplete, kCompleteDefinition);
+
+  const VName module_name_anchor = PrintAnchorVname(anchors[0], file_path_);
+
+  std::cout << Edge(module_name_anchor, kEdgeDefinesBinding, module_vname);
+
+  if (node.Value().Anchors().size() > 1) {
+    const VName module_end_label_anchor =
+        PrintAnchorVname(anchors[1], file_path_);
+    std::cout << Edge(module_end_label_anchor, kEdgeRef, module_vname);
+  }
+}
+
+void KytheFactsExtractor::ExtractModuleInstanceFact(
+    const IndexingFactNode& node) {
+  const auto& anchors = node.Value().Anchors();
+  const VName module_instance_vname(
+      file_path_, CreateModuleInstantiationSignature(anchors[1].Value()));
+  const VName module_instance_anchor = PrintAnchorVname(anchors[1], file_path_);
+  const VName module_type_vname(file_path_,
+                                CreateModuleSignature(anchors[0].Value()));
+  const VName module_type_anchor = PrintAnchorVname(anchors[0], file_path_);
+
+  std::cout << Fact(module_instance_vname, kFactNodeKind, kNodeVariable);
+  std::cout << Fact(module_instance_vname, kFactComplete, kCompleteDefinition);
+
+  std::cout << Edge(module_type_anchor, kEdgeRef, module_type_vname);
+  std::cout << Edge(module_instance_vname, kEdgeTyped, module_type_vname);
+  std::cout << Edge(module_instance_anchor, kEdgeDefinesBinding,
+                    module_instance_vname);
+}
+
+VName PrintAnchorVname(const Anchor& anchor, absl::string_view file_path) {
+  const VName anchor_vname(file_path,
+                           absl::Substitute(R"(@$0:$1)", anchor.StartLocation(),
+                                            anchor.EndLocation()));
+
+  std::cout << Fact(anchor_vname, kFactNodeKind, kNodeAnchor);
+  std::cout << Fact(anchor_vname, kFactAnchorStart,
+                    absl::Substitute(R"($0)", anchor.StartLocation()));
+  std::cout << Fact(anchor_vname, kFactAnchorEnd,
+                    absl::Substitute(R"($0)", anchor.EndLocation()));
+
+  return anchor_vname;
+}
+
+std::string CreateModuleSignature(const absl::string_view module_name) {
```

Comment:
Since `string_view`s are passed by value, there's no need for `const` in their paramaters.